apps/vthread_props/src/main/java/vthread_props/Main.java

```diff
@@ -0,0 +1,232 @@
+/*
+ * Copyright (c) 2025, Red Hat Inc. All rights reserved.
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package vthread_props;
+
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.Properties;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class Main {
+    public static void main(String[] args) {
+        final String r = String.format("%s %s %s %s %s %s",
+                runTest("System Properties Access", Main::testSystemPropertiesAccess),
+                runTest("LocalDateTime.now()", Main::testLocalDateTimeNow),
+                runTest("ZoneId Access", Main::testZoneIdAccess),
+                runTest("Multiple Time Operations", Main::testMultipleTimeOperations),
+                runTest("Virtual Thread Properties", Main::testVirtualThreadProperties),
+                runTest("Process and System Operations", Main::testProcessAndSystemOperations));
+        System.out.println("\n=== RESULT: " + r + " ===");
+    }
+
+    private static boolean testSystemPropertiesAccess() throws Exception {
+        final AtomicReference<Properties> p = new AtomicReference<>();
+        final AtomicReference<Exception> exception = new AtomicReference<>();
+        final Thread vt = Thread.ofVirtual().start(() -> {
+            try {
+                if (!Thread.currentThread().isVirtual()) {
+                    System.out.println("Error: Thread.currentThread().isVirtual() must be true for this test.");
+                }
+                final Properties props = System.getProperties();
+                p.set(props);
+                System.out.println("Accessed " + props.size() + " system properties");
+            } catch (Exception e) {
+                exception.set(e);
+                e.printStackTrace();
+            }
+        });
+        if (!vt.join(Duration.ofMillis(10))) {
```

Comment:
This is expected to rarely fail. Pushing the timeout limit feels of low value since we are not testing the speed of virtual threads here. I would prefer a longer timeout (possibly one second) even if that translates to waiting a bit more in case of failures.

Reply:
My intention is not to measure speed, rather to catch if anything weird starts to happen, like if an operation that used to take 5ms suddenly takes more than 3x that.

Yet it felt like overengineering it to use thresholds.properties.

I bumped it an order of magnitude to 100ms in [e4d6e0e](https://github.com/Karm/mandrel-integration-tests/pull/335/commits/e4d6e0ec78245dda84c665a128dc50497ddc33fa).

I propose that if such timeout is reached, there is an issue to investigate.  It passes on our weakest systems: GHA runners.